app/scripts/services/quota.js

```diff
@@ -0,0 +1,185 @@
+    };
+
+    var getQuotaWarnings = function(resources, quotas, clusterQuotas) {
+      var warnings = [];
+
+      if (!resources || !quotas) {
+        return warnings;
+      }
+
+      _.each(resources, function(resource) {
+        var filteredQuotas = filterQuotasForResource(resource, quotas);
+        var filteredClusterQuotas = filterQuotasForResource(resource, clusterQuotas);
+
+        var rgv = APIService.objectToResourceGroupVersion(resource);
+        var humanizedResource = APIService.kindToResource(resource.kind, true);
+        var humanizedKind = humanizeKind(resource.kind);
+        var quotaKey = "";
+        if (rgv.group) {
+          quotaKey = rgv.group + "/";
+        }
+        quotaKey += rgv.resource;
+        var warningsForQuota = function(quota) {
+          var q = quota.status.total || quota.status;
+          if(q.hard[quotaKey] !== undefined && q.hard[quotaKey] === q.used[quotaKey]) {
+            warnings.push({
+              type: 'error',
+              message: "You are at your quota of " + q.hard[quotaKey] + " " + (q.hard[quotaKey] === "1" ? humanizedKind : humanizedResource) +
+                        " in this project.",
+              details: "You will not be able to create the " + humanizedKind + " '" + resource.metadata.name + "'.",
+              links: [{
+                href: "/project/" + quota.metadata.namespace + "/quota",
+                label: "View Quota",
+                target: "_blank"
+              }]
+            });
+          }
+
+          warnings = warnings.concat(getResourceLimitWarnings(resource, quota));
+        };
+        _.each(filteredQuotas, function(quota) {
+          warningsForQuota(quota);
+        });
+        _.each(filteredClusterQuotas, function(quota) {
+          warningsForQuota(quota);
+        });
+      });
+
+      return warnings;
+    };
+
+    var getLatestQuotaWarnings = function(resources, context) {
+      var result = $q.defer();
+      var quotaWarnings = [];
+      var quotas, clusterQuotas;
+      var remaining = 2;
+      function _checkDone() {
+        if (remaining === 0) {
+          quotaWarnings = getQuotaWarnings(resources, quotas, clusterQuotas);
+          if (!quotaWarnings.length) {
+            result.resolve({quotaWarnings: quotaWarnings});
+          }
+          else {
+            result.reject({quotaWarnings: quotaWarnings});
+          }
+        }
+      }
+
+      // double check using the latest quotas
+      DataService.list("resourcequotas", context, function(quotaData) {
+        quotas = quotaData.by("metadata.name");
+        Logger.log("quotas", quotas);
+        remaining--;
+        _checkDone();
+      });
+
+      DataService.list("appliedclusterresourcequotas", context, function(clusterQuotaData) {
+        clusterQuotas = clusterQuotaData.by("metadata.name");
+        Logger.log("cluster quotas", clusterQuotas);
+        remaining--;
+        _checkDone();
+      });
+
+      return result.promise;
```

Comment:
Maybe only reject the promise if there is an error with the list? Other cases we resolve the promise (either with warnings or an empty array).